proto/proto/types/errors.proto

```diff
@@ -1,14 +1,28 @@
 syntax = "proto3";
 package errors;
 
+// SUBMIT PROVEN TRANSACTION
+// ================================================================================================
+
+// SubmitProvenTransaction errors.
 enum SubmitProvenTransactionError {
     SUBMIT_PROVEN_TRANSACTION_ERROR_UNSPECIFIED = 0;
-    INTERNAL_ERROR = 1;
-    DESERIALIZATION_FAILED = 2;
-    INVALID_TRANSACTION_PROOF = 3;
-    INCORRECT_ACCOUNT_INITIAL_COMMITMENT = 4;
-    INPUT_NOTES_ALREADY_CONSUMED = 5;
-    UNAUTHENTICATED_NOTES_NOT_FOUND = 6;
-    OUTPUT_NOTES_ALREADY_EXIST = 7;
-    TRANSACTION_EXPIRED = 8;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_INTERNAL = 1;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_DESERIALIZATION_FAILED = 2;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_INVALID_TRANSACTION_PROOF = 3;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_INCORRECT_ACCOUNT_INITIAL_COMMITMENT = 4;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_INPUT_NOTES_ALREADY_CONSUMED = 5;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_UNAUTHENTICATED_NOTES_NOT_FOUND = 6;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_OUTPUT_NOTES_ALREADY_EXIST = 7;
+    SUBMIT_PROVEN_TRANSACTION_ERROR_TRANSACTION_EXPIRED = 8;
```

Comment:
Question: why do we need to make them this verbose?

Reply:
Two enum variants, despite being of different `enum`s can share name and causes a collision, which makes the file to fail to build. For that reason, prefixes refering the struct are added ([styling guide here](https://protobuf.dev/programming-guides/style/#enums)). We could use some other, shorter, prefix but couldn't think of anything better than this. In the generated code, the enum (or whatever picked prefix) is removed.

Reply:
Despite the long variant names (maybe we can think of some better prefixes), I like the self-documenting features of having it defined in the protos and the auto-generation that can be achieve for any given language